sw/device/silicon_creator/rom/second_rom.c

```diff
@@ -0,0 +1,321 @@
+#include "sw/device/silicon_creator/lib/drivers/retention_sram.h"
+#include "sw/device/silicon_creator/lib/drivers/rnd.h"
+#include "sw/device/silicon_creator/lib/drivers/rstmgr.h"
+#include "sw/device/silicon_creator/lib/drivers/uart.h"
+#include "sw/device/silicon_creator/lib/drivers/watchdog.h"
+#include "sw/device/silicon_creator/rom/boot_policy.h"
+#include "sw/device/silicon_creator/rom/bootstrap.h"
+#include "sw/device/silicon_creator/rom/second_rom_epmp.h"
+#include "sw/device/silicon_creator/rom/sigverify_keys_rsa.h"
+#include "sw/device/silicon_creator/rom/sigverify_keys_spx.h"
+#include "sw/lib/sw/device/arch/device.h"
+#include "sw/lib/sw/device/base/bitfield.h"
+#include "sw/lib/sw/device/base/csr.h"
+#include "sw/lib/sw/device/base/hardened.h"
+#include "sw/lib/sw/device/base/macros.h"
+#include "sw/lib/sw/device/base/memory.h"
+#include "sw/lib/sw/device/base/stdasm.h"
+#include "sw/lib/sw/device/silicon_creator/base/boot_measurements.h"
+#include "sw/lib/sw/device/silicon_creator/base/sec_mmio.h"
+#include "sw/lib/sw/device/silicon_creator/base/static_critical_version.h"
+#include "sw/lib/sw/device/silicon_creator/boot_data.h"
+#include "sw/lib/sw/device/silicon_creator/cfi.h"
+#include "sw/lib/sw/device/silicon_creator/epmp_state.h"
+#include "sw/lib/sw/device/silicon_creator/error.h"
+#include "sw/lib/sw/device/silicon_creator/rom_print.h"
+#include "sw/lib/sw/device/silicon_creator/shutdown.h"
+#include "sw/lib/sw/device/silicon_creator/sigverify/sigverify.h"
+
+#include "hw/top_darjeeling/sw/autogen/top_darjeeling.h"
+#include "otp_ctrl_regs.h"
+
+/**
+ * Type alias for the ROM_EXT entry point.
+ *
+ * The entry point address obtained from the ROM_EXT manifest must be cast to a
+ * pointer to this type before being called.
+ */
+typedef void rom_ext_entry_point(void);
+
+/**
+ * Table of forward branch Control Flow Integrity (CFI) counters.
+ *
+ * Columns: Name, Initital Value.
+ *
+ * Each counter is indexed by Name. The Initial Value is used to initialize the
+ * counters with unique values with a good hamming distance. The values are
+ * restricted to 11-bit to be able use immediate load instructions.
+
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 6 -n 11 \
+ *     -s 1630646358 --language=c
+ *
+ * Minimum Hamming distance: 6
+ * Maximum Hamming distance: 8
+ * Minimum Hamming weight: 5
+ * Maximum Hamming weight: 8
+ */
+// clang-format off
+#define ROM_CFI_FUNC_COUNTERS_TABLE(X) \
+  X(kCfiRomMain,         0x14b) \
+  X(kCfiRomInit,         0x7dc) \
+  X(kCfiRomVerify,       0x5a7) \
+  X(kCfiRomTryBoot,      0x235) \
+  X(kCfiRomPreBootCheck, 0x43a) \
+  X(kCfiRomBoot,         0x2e2)
+// clang-format on
+
+// Define counters and constant values required by the CFI counter macros.
+CFI_DEFINE_COUNTERS(rom_counters, ROM_CFI_FUNC_COUNTERS_TABLE);
+
+// Life cycle state of the chip.
+lifecycle_state_t lc_state = (lifecycle_state_t)0;
```

Comment:
can you use the symbolic name for 0 here?

Reply:
So it's an invalid state?

Reply:
Yes, it must not be initialised to a valid state. Are you suggesting we should define a `kLcStateInvalid` for that purpose?

Reply:
I think that would be a touch more readable, yes